src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final DateTimeFormatter LOCAL_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String DATA_SPLITTER = " ";
+
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom,
+                                String dateTo) {
+        if (names == null || data == null || dateFrom == null || dateTo == null
+                || names.length == 0 || data.length == 0 || dateFrom.isEmpty()
+                || dateTo.isEmpty()) {
+            return "";
+        }
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, LOCAL_FORMATTER);
+        LocalDate localDateTo = LocalDate.parse(dateTo, LOCAL_FORMATTER);
+        StringBuilder salaryInfo = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo);
+        String[] dataItemSplited = new String[data[0].length()];
+        LocalDate itemDate;
+        String itemEmployerName;
+        int itemWorkedHours;
+        int itemSalaryForHour;
+        int moneyEarned;
```

Comment:
create those variables where you need it, not here

Reply:
ok